components/camel-nats/src/main/java/org/apache/camel/component/nats/NatsProducer.java

```diff
@@ -42,21 +47,42 @@ public NatsEndpoint getEndpoint() {
     }
 
     @Override
-    public void process(Exchange exchange) throws Exception {
+    public void process(Exchange exchange) {
         NatsConfiguration config = getEndpoint().getConfiguration();
         byte[] body = exchange.getIn().getBody(byte[].class);
         if (body == null) {
-            // fallback to use string
-            body = exchange.getIn().getMandatoryBody(String.class).getBytes();
+            try {
+                // fallback to use string
+                body = exchange.getIn().getMandatoryBody(String.class).getBytes();
+            } catch (InvalidPayloadException e) {
+                exchange.setException(e);
```

Comment:
dont catch the exception by let it fail as before as this is the synchronous process method

Reply:
reverted

Reply:
I switched to async callback and added this back, is that the correct approach?

Reply:
yes but it would actually maybe be better to convert to byte[] directly instead of via string as that is double encoding.

Reply:
But that is how the old code was, so it may be something for another PR

Reply:
Isn't the string conversion only a fallback codepath used when `exchange.getIn().getBody(byte[].class)` returns `null`?